slsDetectorServers/mythen3DetectorServer/RegisterDefs.h

```diff
@@ -2,39 +2,166 @@
-#define GET_CYCLES_MSB_REG    			(0x0B << MEM_MAP_SHIFT)
+#define PATTERN_WAIT_1_ADDR_OFST        (0)
+#define PATTERN_WAIT_1_ADDR_MSK         (0x00001FFF << PATTERN_WAIT_1_ADDR_OFST)
+
+/* Pattern Wait 2 RW Register*/
+#define PATTERN_WAIT_2_ADDR_REG         (0x1B0 + BASE_PATTERN_CONTROL)
+
+#define PATTERN_WAIT_2_ADDR_OFST        (0)
+#define PATTERN_WAIT_2_ADDR_MSK         (0x00001FFF << PATTERN_WAIT_2_ADDR_OFST)
+
+/* Pattern Wait Timer 0 64bit RW Register */
+#define PATTERN_WAIT_TIMER_0_LSB_REG    (0x180 + BASE_PATTERN_CONTROL)
+#define PATTERN_WAIT_TIMER_0_MSB_REG    (0x184 + BASE_PATTERN_CONTROL)
+
+/* Pattern Wait Timer 1 64bit RW Register */
+#define PATTERN_WAIT_TIMER_1_LSB_REG    (0x194 + BASE_PATTERN_CONTROL)
+#define PATTERN_WAIT_TIMER_1_MSB_REG    (0x198 + BASE_PATTERN_CONTROL)
+
+/* Pattern Wait Timer 2 64bit RW Register */
+#define PATTERN_WAIT_TIMER_2_LSB_REG    (0x1A8 + BASE_PATTERN_CONTROL)
+#define PATTERN_WAIT_TIMER_2_MSB_REG    (0x1AC + BASE_PATTERN_CONTROL)
+
+/* Pattern Loop 0 Address RW Register */
+#define PATTERN_LOOP_0_ADDR_REG         (0x190 + BASE_PATTERN_CONTROL)
+
+#define PATTERN_LOOP_0_ADDR_STRT_OFST   (0)
+#define PATTERN_LOOP_0_ADDR_STRT_MSK    (0x00001FFF << PATTERN_LOOP_0_ADDR_STRT_OFST)
+#define PATTERN_LOOP_0_ADDR_STP_OFST    (16)
+#define PATTERN_LOOP_0_ADDR_STP_MSK     (0x00001FFF << PATTERN_LOOP_0_ADDR_STP_OFST)
+
+/* Pattern Loop 1 Address RW Register */
+#define PATTERN_LOOP_1_ADDR_REG         (0x1A4 + BASE_PATTERN_CONTROL)
+
+#define PATTERN_LOOP_1_ADDR_STRT_OFST   (0)
+#define PATTERN_LOOP_1_ADDR_STRT_MSK    (0x00001FFF << PATTERN_LOOP_1_ADDR_STRT_OFST)
+#define PATTERN_LOOP_1_ADDR_STP_OFST    (16)
+#define PATTERN_LOOP_1_ADDR_STP_MSK     (0x00001FFF << PATTERN_LOOP_1_ADDR_STP_OFST)
+
+/* Pattern Loop 0 Address RW Register */
+#define PATTERN_LOOP_2_ADDR_REG         (0x1B8 + BASE_PATTERN_CONTROL)
+
+#define PATTERN_LOOP_2_ADDR_STRT_OFST   (0)
+#define PATTERN_LOOP_2_ADDR_STRT_MSK    (0x00001FFF << PATTERN_LOOP_2_ADDR_STRT_OFST)
+#define PATTERN_LOOP_2_ADDR_STP_OFST    (16)
+#define PATTERN_LOOP_2_ADDR_STP_MSK     (0x00001FFF << PATTERN_LOOP_2_ADDR_STP_OFST)
+
+/* Pattern Loop 0 Iteration RW Register */
+#define PATTERN_LOOP_0_ITERATION_REG    (0x18C + BASE_PATTERN_CONTROL) // patnloop
+
+/* Pattern Loop 1 Iteration RW Register */
+#define PATTERN_LOOP_1_ITERATION_REG    (0x1A0 + BASE_PATTERN_CONTROL) // patnloop
+
+/* Pattern Loop 2 Iteration RW Register */
+#define PATTERN_LOOP_2_ITERATION_REG    (0x1B4 + BASE_PATTERN_CONTROL) // patnloop
+
+/* Register of first word */
+#define PATTERN_STEP0_LSB_REG           (0x0 + BASE_PATTERN_RAM)
+#define PATTERN_STEP0_MSB_REG           (0x4 + BASE_PATTERN_RAM)
+
+
+#ifdef VIRTUAL // until firmware is ready ----------------------------------
+
+#define RUN_BUSY_OFST					(0)
+#define RUN_BUSY_MSK      				(0x00000001 << RUN_BUSY_OFST)
+
+/* Set Exptime 64 bit register eEXP = Exp x 25 ns */
+#define SET_EXPTIME_LSB_REG    			(0x08)
```

Comment:
This reg value wouldn't work even if you used it, as you now already have a reg with 0x08 address. I would remove this anyway as I don't think it is used anymore, you write into the set pattern wait time now for acquisition time.